src/graphql/update-user.ts

```diff
@@ -24,31 +55,26 @@ const updateUserEndpoint = {
   },
   async resolve(
     _: any,
-    { avatarUrl, displayName, description, username, email }: any,
+    updateUserArgs: UpdateUserArgs,
     ctx: any
   ) {
-    let user = await getAuthenticatedUser(ctx);
-    if (user.avatarUrl === 'null') {
-      user.avatarUrl = null;
-    } else {
-      user.avatarUrl = avatarUrl;
+    const user = await getAuthenticatedUser(ctx);
+    const validatedUpdate = await updateUserSchema.validate(updateUserArgs, {stripUnknown: true});
+    const existingUser = await getByUsernameOrEmailExcludingId(validatedUpdate.username, validatedUpdate.email, user.id);
+    if (existingUser) {
+      if (existingUser.email === validatedUpdate.email) {
+        throw 'Account with given email already exists';
+      }
+      if (existingUser.username === validatedUpdate.username) {
+        throw 'Username is already taken';
+      }
```

Comment:
make sure, that we return `400` response with the proper error message.

Reply:
we are getting an error even with 200 response code